src/tasklet/tasklet_ptr.rs

```diff
@@ -0,0 +1,178 @@
+pub(crate) struct TaskletPtr {
+    /// Pointer to the `Tasklet<T, C>` structure.
+    ptr: *const (),
+    /// Tasklet virtual table.
+    vtable: &'static TaskletVTable,
+}
+
+unsafe impl Sync for TaskletPtr {}
+unsafe impl Send for TaskletPtr {}
+
+impl TaskletPtr {
+    /// Creates new pointer
+    ///
+    /// * `ptr` - Pointer to memory where tasklet is allocated.
+    pub(crate) fn new<T: 'static, C>(ptr: *const ()) -> Self {
+        TaskletPtr {
+            ptr,
+            vtable: tasklet_vtable::<T, C>(),
+        }
+    }
+
+    /// Returns tasklet name.
+    #[inline(always)]
+    pub(crate) fn get_name(&self) -> &'static str {
+        (self.vtable.get_name)(self.ptr)
+    }
+
+    /// Returns tasklet status.
+    #[inline(always)]
+    pub(crate) fn get_status(&self) -> TaskStatus {
+        (self.vtable.get_status)(self.ptr)
+    }
+
+    /// Sets tasklet status.
+    ///
+    /// * `status` - New tasklet status.
+    #[inline(always)]
+    pub(crate) fn set_status(&self, status: TaskStatus) {
+        (self.vtable.set_status)(self.ptr, status)
+    }
+
+    /// Returns last execution time.
+    #[inline(always)]
+    pub(crate) fn get_last_execution_time(&self) -> TimerInstantU64<1_000_000> {
+        (self.vtable.get_last_execution_time)(self.ptr)
+    }
+
+    /// Sets last execution time.
+    ///
+    /// * `time` - Last execution time.
+    #[inline(always)]
+    pub(crate) fn set_last_execution_time(&self, time: TimerInstantU64<1_000_000>) {
+        (self.vtable.set_last_execution_time)(self.ptr, time)
+    }
+
+    /// Checks if there are any more data for the tasklet to process.
+    #[inline(always)]
+    pub(crate) fn has_work(&self) -> bool {
+        (self.vtable.has_work)(self.ptr)
+    }
+
+    /// Executes task.
+    #[inline(always)]
+    pub(crate) fn execute(&self) {
+        (self.vtable.execute)(self.ptr)
+    }
+}
+
+impl Ord for TaskletPtr {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.get_last_execution_time()
+            .cmp(&other.get_last_execution_time())
+            .reverse()
+    }
+}
+
+impl PartialOrd for TaskletPtr {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        self.get_last_execution_time()
+            .partial_cmp(&other.get_last_execution_time())
+            .map(|ordering| ordering.reverse())
+    }
+}
+
+impl Eq for TaskletPtr {}
+
+impl PartialEq for TaskletPtr {
+    fn eq(&self, other: &Self) -> bool {
+        self.get_last_execution_time()
+            .eq(&other.get_last_execution_time())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::task::Task;
+    use crate::tasklet::{Tasklet, TaskletConfig};
+
+    fn create_tasklet() -> Tasklet<u8, ()> {
+        let tasklet_config = TaskletConfig { name: "TaskName" };
+
+        Tasklet::<u8, ()>::new(tasklet_config)
+    }
+
+    fn create_tasklet_ptr(tasklet: &Tasklet<u8, ()>) -> TaskletPtr {
+        let ptr = tasklet as *const Tasklet<u8, ()> as *const ();
```

Comment:
fix double-cast

Reply:
also; why is this function (along with create_tasklet) repeated here?

Reply:
fixed double cas

By `repeated` do you mean that they are also present in the `tasklet_handle.rs`? Those are tests helper functions. We could (and we will probably have to in future) create some test harness with helpers and alike, but for now it's not that necessary.